docs/testing/3-subcomponents.md

```diff
@@ -0,0 +1,38 @@
+# Testing SubComponents
+
+As subcomponents extend from JavaFX nodes mocking them destroys their functionality making them useless and prevents them from being rendered.
+Spying has similar issues. Another problem with subcomponents is that they often require multiple dependencies like services themselves.
+
+Therefor the best way of testing a subcomponent is by creating a field inside the controller test and annotating it with `@InjectMocks` so that all the dependencies are injected into it as well.
```

Comment:
```suggestion
Therefore the best way of testing a subcomponent is by creating a field inside the controller test and annotating it with `@InjectMocks` so that all the dependencies are injected into it as well.
```